File: frontend/lib/models/equipment_type.dart
enum EquipmentType {
  swing('Salıncak', 'SWING'),
  slide('Kaydırak', 'SLIDE'),
  teeterTotter('Tahtarevalli', 'TEETER_TOTTER'),
  spinner(
    'Dönen Oyun Elemanı',
    'SPINNER',
  ), //TODO ŞUNUN DOĞRU ADINI BUL ÇOK GENEL BİR TABİR İSİM BULAMIYORUM
  climber('Tırmanma Elemanı', 'CLIMBER');

  final String displayName;
  final String apiValue;

  const EquipmentType(this.displayName, this.apiValue);

  static EquipmentType fromApiValue(String value) {
    return EquipmentType.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen Cisim Türü: $value'),
    );
  }
}

File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/EquipmentType.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum EquipmentType {
    SWING("Salıncak"),
    SLIDE("Kaydırak"),
    TEETER_TOTTER("Tahtarevalli"),
    SPINNER("Dönen Oyun Elemanı"),
    CLIMBER("Tırmanma Elemanı");

    private final String displayName;

    EquipmentType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}
